javatests/com/google/copybara/WorkflowTest.java

```diff
@@ -1211,6 +1213,55 @@ public void checkLastRevStatus_change_request() throws Exception {
     }
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  public void givenLastRevFlagInfoCommandUsesIt()
+          throws IOException, RepoException, ValidationException {
+    Path originPath = Files.createTempDirectory("origin");
+    Path destinationPath = Files.createTempDirectory("destination");
+    GitRepository origin = initScratchRepo( /*verbose=*/true, originPath, getGitEnv());
+    GitRepository destination = initScratchRepo( /*verbose=*/true, destinationPath, getGitEnv());
+
+    String config = "core.workflow("
+            + "    name = '" + "default" + "',"
+            + "    origin = git.origin( url = 'file://" + origin.getWorkTree() + "', ref = 'master' ),\n"
+            + "    destination = git.destination( url = 'file://" + destination.getWorkTree() + "'),"
+            + "    authoring = " + authoring + ","
+            + "    mode = '" + WorkflowMode.ITERATIVE + "',"
+            + ")\n";
+
+    Files.write(originPath.resolve("foo.txt"), "not important".getBytes(UTF_8));
+    origin.add().files("foo.txt").run();
+    origin.commit("Foo <foo@bara.com>", ZonedDateTime.now(), "not important");
+    String firstCommit = origin.parseRef("HEAD");
+
+    Files.write(destinationPath.resolve("foo.txt"), "not important".getBytes(UTF_8));
+    destination.add().files("foo.txt").run();
+    destination.commit("Foo <foo@bara.com>", ZonedDateTime.now(), "not important");
+
+    Files.write(originPath.resolve("foo.txt"), "foo".getBytes(UTF_8));
+    origin.add().files("foo.txt").run();
+    origin.commit("Foo <foo@bara.com>", ZonedDateTime.now(), "change1");
+
+    options.setWorkdirToRealTempDir();
+    // Pass custom HOME directory so that we run an hermetic test and we
+    // can add custom configuration to $HOME/.gitconfig.
+    options.setEnvironment(GitTestUtil.getGitEnv());
+    options.setHomeDir(Files.createTempDirectory("home").toString());
+    options.gitDestination.committerName = "Foo";
+    options.gitDestination.committerEmail = "foo@foo.com";
+    options.workflowOptions.checkLastRevState = true;
+    options.setLastRevision(firstCommit);
+
+    final Info<Revision> info = (Info<Revision>) loadConfig(config).getMigration("default").getInfo();
+    final List<String> commitMessages =
```

Comment:
There's actually two levels of nesting there... `migrationReferences` and `getAvailableToMigrate`. I just code in Scala in my day to day so streams were easier. Is there a similar nifty thing for two levels of iterables?